src/main/java/de/tum/in/www1/artemis/service/messaging/DistributedInstanceMessageSendService.java

```diff
@@ -135,6 +135,18 @@ public void sendParticipantScoreSchedule(Long exerciseId, Long participantId, Lo
         sendMessageDelayed("participant-score-schedule", exerciseId, participantId, resultId);
     }
 
+    @Override
+    public void sendProgressUpdateForExercise(Long exerciseId, Long userId) {
+        log.info("Sending update learning goal progress for exercise {} and user {}.", exerciseId, userId);
+        sendMessageDelayed("schedule-progress-exercise", exerciseId, userId);
```

Comment:
Should we maybe move them into constants instead of creating these strings in multiple places?
Simply adding more and more of these string duplications will only make the problem worse

Reply:
Yes, we should think about refactoring this. In any case, this should be done in a separate PR.

Reply:
I've added an enum for the different message topics.